### DISCUSSION TITLE
Updating notebook packages without running it

### DISCUSSION THREAD

**Discussion by cstjean** [fonsp/Pluto.jl#3060] [Questions about using Pluto] [answered]
I have 5-6 notebooks using heavy dependencies, like ModelingToolkit.jl and Plots.jl (around 200 dependencies total). Updating any of the notebooks turns out to be a major endeavour, because it means:

1. Load the notebook
2. I want to update the packages right away, but I can't
![{72B4428D-DD08-4ADB-AF2A-0D86C94E40D8}](https://github.com/user-attachments/assets/4b05a48c-2304-4c27-9d1c-d4bace0d961f)
There's no button for updating!

So I have to:
1. Click on "Run notebook code". This will typically take ages (10 minutes?), because the Pkg manager typically has flushed the precompilation files out of the cache for those packages.
2. Interrupt it so it doesn't run everything
3. Update the packages, wait for precompilation, which again takes 10 minutes.
4. Click on restart

The whole thing takes about 20 minutes per notebook. I think part of the problem is that ModelingToolkit etc. is so huge that it easily overwhelms the Pkg cache. Or at least, that's my theory.

At any rate, in the REPL, there is no problem. Activating an old environment does not entail precompiling that old environment. I can just call `]up`, and Julia will only precompile the new environment, and since all environments will `]up` to the same version of ModelingToolkit (at least, usually), then that work gets to be reused and the Pkg cache issue goes away.

So I guess what I'm asking is: can we please be allowed to update packages _without_ clicking on "Run notebook code"? I know that I could `activate_notebook_environment` at the REPL, but that's not very convenient.

**Comment by greimel** [comment]
I agree that it would be nice if this were easier.

I usually add a cell with "using Pkg; Pkg.activate(temp = true)" before "running notebook code". That erases the environment, and yields a quick error. Then remove that cell and run again.

Note that there is a [script for that purpose in the computational-thinking repo](https://github.com/mitmath/computational-thinking/blob/Fall24/tools/update_notebook_packages.jl), it uses `Pluto.update_notebook_environment`.

**Comment by fonsp** [comment, marked as answer]
Let's do it! I made https://github.com/fonsp/Pluto.jl/issues/3079
